presto-atop/src/main/java/com/facebook/presto/atop/AtopProcessFactory.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.atop;
+
+import com.facebook.presto.spi.PrestoException;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+import javax.inject.Inject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.NoSuchElementException;
+import java.util.concurrent.TimeUnit;
+
+import static com.facebook.presto.atop.AtopErrorCode.ATOP_CANNOT_START_PROCESS_ERROR;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class AtopProcessFactory
+        implements AtopFactory
+{
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormat.forPattern("YYYYMMdd");
+    private final String executablePath;
+    private final DateTimeZone timeZone;
+
+    @Inject
+    public AtopProcessFactory(AtopConnectorConfig config)
+    {
+        this.executablePath = config.getExecutablePath();
+        this.timeZone = config.getDateTimeZone();
+    }
+
+    @Override
+    public Atop create(AtopTable table, DateTime date)
+    {
+        checkArgument(date.getZone().equals(timeZone), "Split date (%s) is not in the local timezone (%s)", date.getZone(), timeZone);
+
+        ProcessBuilder processBuilder = new ProcessBuilder(executablePath);
```

Comment:
Consider using Airlift Command library to manage the process (https://github.com/airlift/command/blob/master/src/main/java/io/airlift/command/Command.java).  There are many edge cases in managing process from Java.